File: google-play-api-plugin/src/main/java/com/dipien/google/play/api/GooglePlayPublisherPlugin.java
package com.dipien.google.play.api;

import com.dipien.google.play.api.task.ListTracksTask;
import com.dipien.google.play.api.task.MetadataVerificationTask;
import com.dipien.google.play.api.task.PromoteFromAlphaToBetaTask;
import com.dipien.google.play.api.task.PromoteFromAlphaToProductionTask;
import com.dipien.google.play.api.task.PromoteFromBetaToProductionTask;
import com.dipien.google.play.api.task.PromoteFromInternalToAlphaTask;
import com.dipien.google.play.api.task.PromoteFromInternalToBetaTask;
import com.dipien.google.play.api.task.PromoteFromInternalToProductionTask;
import com.dipien.google.play.api.task.PublishApkTask;
import com.dipien.google.play.api.task.PublishBundleTask;
import com.dipien.google.play.api.task.PublishMetadataTask;
import com.dipien.google.play.api.task.ResumeStagedRolloutTask;
import com.dipien.google.play.api.task.UploadBundleToInternalAppSharingTask;
import com.dipien.google.play.api.task.CompleteStagedRollout;
import com.dipien.google.play.api.task.HaltStagedRolloutTask;
import com.dipien.google.play.api.task.IncreaseStagedRolloutTask;
import com.dipien.google.play.api.task.ListApksTask;
import com.dipien.google.play.api.task.ListBundlesTask;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class GooglePlayPublisherPlugin implements Plugin<Project> {
	
	public static final String EXTENSION_NAME = "googlePlay";
	
	@Override
	public void apply(Project project) {
		
		project.getExtensions().create(EXTENSION_NAME, GooglePlayPublisherPluginExtension.class, project);
		
		
		project.getTasks().create("googlePlayVerifyMetadata", MetadataVerificationTask.class);
		project.getTasks().create("googlePlayPublishMetadata", PublishMetadataTask.class);
		project.getTasks().create("googlePlayListAPKs", ListApksTask.class);
		project.getTasks().create("googlePlayListBundles", ListBundlesTask.class);
		project.getTasks().create("googlePlayPublishAPK", PublishApkTask.class);
		project.getTasks().create("googlePlayPublishBundle", PublishBundleTask.class);
		project.getTasks().create("googlePlayUploadBundleToInternalAppSharing", UploadBundleToInternalAppSharingTask.class);
		project.getTasks().create("googlePlayPromoteFromInternalToAlpha", PromoteFromInternalToAlphaTask.class);
		project.getTasks().create("googlePlayPromoteFromInternalToBeta", PromoteFromInternalToBetaTask.class);
		project.getTasks().create("googlePlayPromoteFromInternalToProduction", PromoteFromInternalToProductionTask.class);
		project.getTasks().create("googlePlayPromoteFromAlphaToBeta", PromoteFromAlphaToBetaTask.class);
		project.getTasks().create("googlePlayPromoteFromAlphaToProduction", PromoteFromAlphaToProductionTask.class);
		project.getTasks().create("googlePlayPromoteFromBetaToProduction", PromoteFromBetaToProductionTask.class);
		project.getTasks().create("googlePlayIncreaseStagedRollout", IncreaseStagedRolloutTask.class);
		project.getTasks().create("googlePlayHaltStagedRollout", HaltStagedRolloutTask.class);
		project.getTasks().create("googlePlayResumeStagedRollout", ResumeStagedRolloutTask.class);
		project.getTasks().create("googlePlayCompleteStagedRollout", CompleteStagedRollout.class);
		project.getTasks().create("googlePlayListTracks", ListTracksTask.class);
	}

}

File: google-play-api-plugin/src/main/java/com/dipien/google/play/api/GooglePlayPublisherPlugin.kt
package com.dipien.google.play.api

import com.dipien.google.play.api.task.CompleteStagedRolloutTask
import com.dipien.google.play.api.task.HaltStagedRolloutTask
import com.dipien.google.play.api.task.IncreaseStagedRolloutTask
import com.dipien.google.play.api.task.ListApksTask
import com.dipien.google.play.api.task.ListBundlesTask
import com.dipien.google.play.api.task.ListTracksTask
import com.dipien.google.play.api.task.MetadataVerificationTask
import com.dipien.google.play.api.task.PromoteFromAlphaToBetaTask
import com.dipien.google.play.api.task.PromoteFromAlphaToProductionTask
import com.dipien.google.play.api.task.PromoteFromBetaToProductionTask
import com.dipien.google.play.api.task.PromoteFromInternalToAlphaTask
import com.dipien.google.play.api.task.PromoteFromInternalToBetaTask
import com.dipien.google.play.api.task.PromoteFromInternalToProductionTask
import com.dipien.google.play.api.task.PublishBundleTask
import com.dipien.google.play.api.task.PublishMetadataTask
import com.dipien.google.play.api.task.ResumeStagedRolloutTask
import com.dipien.google.play.api.task.UploadBundleToInternalAppSharingTask
import org.gradle.api.Plugin
import org.gradle.api.Project

class GooglePlayPublisherPlugin : Plugin<Project> {

    companion object {
        const val EXTENSION_NAME = "googlePlay"
    }

    override fun apply(project: Project) {
        project.extensions.create(EXTENSION_NAME, GooglePlayPublisherPluginExtension::class.java, project)
        project.tasks.create("googlePlayVerifyMetadata", MetadataVerificationTask::class.java)
        project.tasks.create("googlePlayPublishMetadata", PublishMetadataTask::class.java)
        project.tasks.create("googlePlayListAPKs", ListApksTask::class.java)
        project.tasks.create("googlePlayListBundles", ListBundlesTask::class.java)
        project.tasks.create("googlePlayPublishBundle", PublishBundleTask::class.java)
        project.tasks.create("googlePlayUploadBundleToInternalAppSharing", UploadBundleToInternalAppSharingTask::class.java)
        project.tasks.create("googlePlayPromoteFromInternalToAlpha", PromoteFromInternalToAlphaTask::class.java)
        project.tasks.create("googlePlayPromoteFromInternalToBeta", PromoteFromInternalToBetaTask::class.java)
        project.tasks.create("googlePlayPromoteFromInternalToProduction", PromoteFromInternalToProductionTask::class.java)
        project.tasks.create("googlePlayPromoteFromAlphaToBeta", PromoteFromAlphaToBetaTask::class.java)
        project.tasks.create("googlePlayPromoteFromAlphaToProduction", PromoteFromAlphaToProductionTask::class.java)
        project.tasks.create("googlePlayPromoteFromBetaToProduction", PromoteFromBetaToProductionTask::class.java)
        project.tasks.create("googlePlayIncreaseStagedRollout", IncreaseStagedRolloutTask::class.java)
        project.tasks.create("googlePlayHaltStagedRollout", HaltStagedRolloutTask::class.java)
        project.tasks.create("googlePlayResumeStagedRollout", ResumeStagedRolloutTask::class.java)
        project.tasks.create("googlePlayCompleteStagedRollout", CompleteStagedRolloutTask::class.java)
        project.tasks.create("googlePlayListTracks", ListTracksTask::class.java)
    }
}
